packages/xforms-engine/test/parse/xpath/dependency-analysis.test.ts

```diff
@@ -0,0 +1,337 @@
+import { describe, expect, it } from 'vitest';
+import { resolveDependencyNodesets } from '../../../src/parse/xpath/dependency-analysis.ts';
+
+describe('Dependency analysis', () => {
```

Comment:
I have tried hard to think of missing cases but this is feeling really complete to me. Maybe something with a known function? I would expect any arguments that are paths to be identified as dependencies and resolved I think but https://github.com/getodk/web-forms/pull/166/files#diff-cb3208eef759b81b33d971b0f6ccea7c8f2c0c041fdc98b2be1d9b515855cc13R163 makes me question that.

Reply:
Same response as there! I don't know if `if()` is what you're looking for. But it's specifically covered because I wanted to make sure at least one FunctionCall was included, and because that function in particular has been called out in the past for its conditional Argument evaluation. The point of including it is that all of the Arguments are analyzed, regardless of the branch they'll hit when evaluated.